### PR TITLE
fixed:#511 search not working fixed

### PR DESCRIPTION
**Pull request by Sinharitik589**
## Related Issues

#511 

## Checklist

- [x] I have tested these changes
- [ ] I have updated the relevant documentation
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the codebase
- [x] My changes generate no new warnings or errors
- [x] The title of my pull request is clear and descriptive

### CODE REVIEW

File: frontend/src/components/AutoCompleteAsync/index.jsx [hunk 1]
```diff
@@ -57,13 +57,11 @@ export default function AutoCompleteAsync({
   }, [debouncedValue]);
 
   const onSearch = (searchText) => {
-    if (searchText && searchText != '') {
```
Comment (Sinharitik589) [inline frontend/src/components/AutoCompleteAsync/index.jsx:60]: No checks needed here since we are already checking debouncedValue everywhere

Comment (salahlalami) [reply frontend/src/components/AutoCompleteAsync/index.jsx:60]: revert this line of code , this will break when we try to update

**Review by salahlalami — CHANGES_REQUESTED** [review]

**Comment by polymahh** [comment]
@Sinharitik589 revert the changes that salah want
i see the solution u did worked but i think it keeps recalling onSearch for the same value like "s"
i think we need to reference the value if it is the same there is no need to call onSearch

**Comment by Sinharitik589** [comment]
> @Sinharitik589 revert the changes that salah want i see the solution u did worked but i think it keeps recalling onSearch for the same value like "s" i think we need to reference the value if it is the same there is no need to call onSearch

The main issue that i encountered was that whenever there is one letter suppose you search z and then you defocus and then again you enter the same search term it will show the loading text as currently we only set Current value when searchText is not '' so when we debounce the state doesnt change on entering the same value z again hence the component doesn't rerender ,so the text keeps saying isLoading .  So the second issue will not be addresed , only the part where it was searching based on the company name , manager name etc will be addresed

**Comment by polymahh** [comment]
@Sinharitik589 pls test this i didnt test it good enough so am not sure about it 
image

**Comment by Sinharitik589** [comment]
> @Sinharitik589 pls test this i didnt test it good enough so am not sure about it image

This also works well , but i dont see any thing breaking with
image

i tested the update functionality its working fine , so i dont think any necessity to use an extra ref here , since state is not updating as the component is not mounting so any way valToSearch persists anyway

**Comment by Sinharitik589** [comment]
> @Sinharitik589 pls test this i didnt test it good enough so am not sure about it image

It works well with this , but we don't need an extra ref for that purpose we can check that with the existing state valToSearch 
image

**Review by salahlalami — APPROVED** [review]

**Comment by salahlalami** [comment]
@Sinharitik589 share your pr in slack for review

**Review by polymahh — APPROVED** [review]

**Pull request merged**